osu.Game/Screens/Ranking/UserTagControl.cs

```diff
@@ -0,0 +1,537 @@
+                                    {
+                                        Alpha = 0,
+                                        Size = new Vector2(18),
+                                    }
+                                }
+                            }
+                        }
+                    }
+                });
+
+                TooltipText = UserTag.Description;
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                const double transition_duration = 300;
+
+                voteCount.BindValueChanged(_ =>
+                {
+                    voteCountText.Text = voteCount.Value.ToLocalisableString();
+                    confirmed.Value = voteCount.Value >= 10;
+                }, true);
+                voted.BindValueChanged(v =>
+                {
+                    if (v.NewValue)
+                    {
+                        voteBackground.FadeColour(colours.Lime3, transition_duration, Easing.OutQuint);
+                        voteCountText.FadeColour(Colour4.Black, transition_duration, Easing.OutQuint);
+                    }
+                    else
+                    {
+                        voteBackground.FadeColour(colours.Gray2, transition_duration, Easing.OutQuint);
+                        voteCountText.FadeColour(Colour4.White, transition_duration, Easing.OutQuint);
+                    }
+                }, true);
+                confirmed.BindValueChanged(c =>
+                {
+                    if (c.NewValue)
+                    {
+                        mainBackground.FadeColour(colours.Lime1, transition_duration, Easing.OutQuint);
+                        tagNameText.FadeColour(Colour4.Black, transition_duration, Easing.OutQuint);
+                        FadeEdgeEffectTo(0.5f, transition_duration, Easing.OutQuint);
+                    }
+                    else
+                    {
+                        mainBackground.FadeColour(colours.Gray4, transition_duration, Easing.OutQuint);
+                        tagNameText.FadeColour(Colour4.White, transition_duration, Easing.OutQuint);
+                        FadeEdgeEffectTo(0f, transition_duration, Easing.OutQuint);
+                    }
+                }, true);
+                FinishTransforms(true);
+
+                Action = () =>
+                {
+                    if (requestInFlight != null)
+                        return;
+
+                    spinner.Show();
+
+                    APIRequest request;
+
+                    switch (voted.Value)
+                    {
+                        case true:
+                            var removeReq = new RemoveBeatmapTagRequest(beatmap.Value.BeatmapInfo.OnlineID, UserTag.Id);
+                            removeReq.Success += () =>
+                            {
+                                voteCount.Value -= 1;
+                                voted.Value = false;
```

Comment:
I guess the only weird thing I've seen is that a tag returning to 0 count doesn't disappear and instead stays visible with a `0`. Maybe fine, not sure.

Reply:
Applied in https://github.com/ppy/osu/pull/32221/commits/3d4dd8507723fcd3a048442834336486debb9732